### PR TITLE
PIMS-61 Forcing DB migration on API startup

### PR DESCRIPTION
**Pull request by Fosol**
## Description 

Continued work on the production pipeline.

- Updating dependencies
- Forcing DB migration on API startup.  This isn't ideal, but the amount of effort to do it in the DevOps pipeline is becoming ridiculous.
- Removing HealthCheck UI from API

### CODE REVIEW

File: backend/api/Startup.cs [hunk 1]
```diff
@@ -238,9 +238,10 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IApiVers
             if (!env.IsProduction())
             {
                 app.UseDatabaseErrorPage();
-                app.UpdateDatabase<Startup>();
             }
 
+            app.UpdateDatabase<Startup>();
```
Comment (Fosol) [inline backend/api/Startup.cs:243]: Forcing DB migration on startup...

File: openshift/templates/pims-api/build.yaml [hunk 2]
```diff
@@ -50,6 +50,7 @@ objects:
           from:
             kind: ImageStreamTag
             name: ${DOTNET_BUILDER_IMAGE}:${DOTNET_BUILDER_TAG}
+          # scripts: https://raw.githubusercontent.com/bcgov/PIMS/dev/.s2i/bin
```
Comment (Fosol) [inline openshift/templates/pims-api/build.yaml:53]: This is required to resolve the user permission issue when attempting to run the `dotnet ef database update` command on deployment.  There are other issues that stop the hook from working however.

File: openshift/templates/pims-api/deploy.yaml [hunk 3]
```diff
@@ -160,16 +160,16 @@ objects:
...
-            execNewPod:
-              command:
-                - bash
-                - "-c"
-                - >-
-                  dotnet ef database update --project=api --context=pimsContext
-                  --verbose
-              containerName: ${APP_NAME}-${COMP_NAME}-${ENV_NAME}
-            failurePolicy: Abort
+          # pre:
```
Comment (Fosol) [inline openshift/templates/pims-api/deploy.yaml:163]: The hook still doesn't work so I've commented it out.

File: openshift/templates/pims-db/mssql-deploy.yaml [hunk 4]
```diff
@@ -211,7 +211,7 @@ parameters:
   - name: DB_PASSWORD
     displayName: Microsoft SQL Server Password
     description: Microsoft SQL Server sa password
-    from: '[\w]{12}'
+    from: '[\w]{12}_1Ab!'
```
Comment (Fosol) [inline openshift/templates/pims-db/mssql-deploy.yaml:214]: Ensuring minimum password complexity.

File: backend/api/appsettings.Local.json [hunk 5]
```diff
@@ -2,20 +2,6 @@
   "HealthChecks": {
     "Port": "5000"
   },
-  "HealthChecksUI": {
```
Comment (devinleighsmith) [inline backend/api/appsettings.Local.json:5]: this doesn't seem related to the title of the PR, intended?

Comment (Fosol) [reply backend/api/appsettings.Local.json:5]: I did include it in the description of the PR.  I am working on the prod pipeline.  While making sure the DB migration occurs during a deploy I also cleaned up some of the code in the related method.  Additionally I've updated the dependencies and removed one that isn't required anymore.
